packages/discord.js/src/structures/GuildMember.js

```diff
@@ -236,12 +236,12 @@ class GuildMember extends Base {
   }
 
   /**
-   * The nickname of this member, or their username if they don't have one
+   * The nickname of this member, or their user display name if they don't have one
    * @type {?string}
    * @readonly
    */
   get displayName() {
-    return this.nickname ?? this.user.username;
+    return this.nickname ?? this.user.discriminator === '0' ? this.user.globalName : this.user.username;
```

Comment:
```suggestion
    return this.nickname ?? this.user.globalName ?? this.user.username;
```
`globalName` can be null and if the user is using the new system and doesn't have one this would return null which isnt expected

Reply:
Thanks! I have added `User#displayName`.